source/extensions/transport_sockets/alts/tsi_socket.cc

```diff
@@ -193,14 +258,7 @@ Network::IoResult TsiSocket::doRead(Buffer::Instance& buffer) {
 
   if (handshake_complete_) {
     ASSERT(frame_protector_);
-
-    uint64_t read_size = raw_read_buffer_.length();
-    ENVOY_CONN_LOG(debug, "TSI: unprotecting buffer size: {}", callbacks_->connection(),
-                   raw_read_buffer_.length());
-    tsi_result status = frame_protector_->unprotect(raw_read_buffer_, buffer);
-    ENVOY_CONN_LOG(debug, "TSI: unprotected buffer left: {} result: {}", callbacks_->connection(),
-                   raw_read_buffer_.length(), tsi_result_to_string(status));
-    result.bytes_processed_ = read_size - raw_read_buffer_.length();
+    return repeatReadAndUnprotect(buffer, result);
```

Comment:
The ENVOY_CONN_LOG on line 264 is no longer hit due to early returns like this one.

Reply:
Good point. I moved it inside `repeatReadAndUnprotect`.